src/dolphin/utils.py

```diff
@@ -557,3 +557,110 @@ def flatten(list_of_lists: Iterable[Iterable[Any]]) -> chain[Any]:
 
 def _format_date_pair(start: DateOrDatetime, end: DateOrDatetime, fmt="%Y%m%d") -> str:
     return f"{start.strftime(fmt)}_{end.strftime(fmt)}"
+
+
+def prepare_geometry(
```

Comment:
this function might make more sense in the `stitching` module, but we can revisit later